File: sicpa-web/src/app/models/department-employee-dto.ts
import {DepartmentDto} from "./department-dto";
import {EmployeeDto} from "./employee-dto";
import {GenericDto} from "./generic-dto";

export class DepartmentEmployeeDto extends GenericDto {
  id: number;
  department: DepartmentDto = new DepartmentDto();
  employee: EmployeeDto = new EmployeeDto();

  constructor() {
    super();
  }
}

File: sicpa-root/sicpa-dto/src/main/java/com/sicpa/demo/dto/DepartmentEmployeeDto.java
package com.sicpa.demo.dto;

import java.io.Serializable;

import javax.validation.constraints.NotNull;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class DepartmentEmployeeDto extends BaseDto implements Serializable {
    private static final long serialVersionUID = 1L;
    private Long id;
    @NotNull
    private DepartmentDto department;
    @NotNull
    private EmployeeDto employee;
}